angular/run/routes.run.js

```diff
@@ -1,17 +1,17 @@
-export function RoutesRun($rootScope, $state, $auth) {
+export function RoutesRun($state, $transitions) {
     'ngInject';
 
+    let requiresAuthCriteria = {
+        to: ($state) => $state.data && $state.data.auth
+    };
 
-    let deregisterationCallback =  $rootScope.$on("$stateChangeStart", function(event, toState) {
-
-        if (toState.data && toState.data.auth) {
-            /*Cancel going to the authenticated state and go back to the login page*/
-            if (!$auth.isAuthenticated()) {
-                event.preventDefault();
-                return $state.go('app.login');
-            }
+    let redirectToLogin = ['$auth', '$transition$', ($auth) => {
```

Comment:
It doesn't work if i move them to the top, they're special injectables for the onBefore method, note that the `$transitions` it's not the same as `$transition$ used here, the `$transition$` returns the current transition

[http://angular-ui.github.io/ui-router/feature-1.0/interfaces/transition.ihookregistry.html#onbefore](See the docs)